src/data/DataStore.ts

```diff
@@ -178,6 +178,10 @@ class DataStore {
 
     private _calcDimNameToIdx = createHashMap<DimensionIndex, DimensionName>();
 
+    // TODO (immutableMode): make immutableMode configurable, similar to option params replaceMerge, notMerge, lazyUpdate
+    // - https://echarts.apache.org/en/api.html#echartsInstance.setOption
+    private immutableMode = true;
```

Comment:
I still need to figure out how to initialize DataStore with the immutableMode property correctly, passing immutableMode in [OptionManager](https://github.com/sjcobb/echarts/pull/1/files#diff-f9b6d53ea14ef369f7f597714b68be9b8fa4f9b05831e2b78ef2b8d84c3c3972) was more straightforward.

I also need to test the changes separately, which helps the most and should bypassing `cloneChunk` and `_cloneIndices` be handled in a separate PR?